samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -40,20 +42,44 @@ static int get_report(struct usb_setup_packet *setup, s32_t *len, u8_t **data)
 		if (request_value[1] == 0x03) {
 			/* Request for feature report */
 			if (request_value[0] == REPORT_ID_USER_CONFIG) {
-				/* Event ID and recipient are retained from latest request from host */
-				frame.status = atomic_get(&forward_status);
-				frame.event_data_len = 0;
+				size_t length = *len;
+				u8_t *buffer = *data;
 
-				int pos = config_channel_report_fill(*data, REPORT_SIZE_USER_CONFIG, &frame, true);
+				struct config_channel_frame frame;
+				int pos = config_channel_report_parse(buffer, length, &frame, true);
 				if (pos < 0) {
-					LOG_WRN("Could not set report");
+					LOG_WRN("Could not parse report");
 					return pos;
 				}
 
+				bool is_fetch = frame.status == FORWARD_STATUS_FETCH;
+				if (is_fetch) {
+					frame.status = fetch_done ? FORWARD_STATUS_SUCCESS : FORWARD_STATUS_PENDING;
+					pos = config_channel_report_fill(buffer, length,
+									 &frame, true);
+					if (pos < 0) {
+						LOG_WRN("Could not set report");
+						return pos;
+					}
+
+					if (fetch_done) {
+						memcpy(&buffer[pos], fetched_data, sizeof(fetched_data));
+					}
+				} else {
+					/* Event ID and recipient are retained from latest request from host */
+					frame.status = atomic_get(&forward_status);
+					frame.event_data_len = 0;
+
+					pos = config_channel_report_fill(buffer, REPORT_SIZE_USER_CONFIG, &frame, true);
+					if (pos < 0) {
+						LOG_WRN("Could not set report");
+						return pos;
+					}
+				}
+
 				pos += frame.event_data_len;
 				__ASSERT_NO_MSG(pos <= REPORT_SIZE_USER_CONFIG);
-
-				*len = pos;
+				length = pos;
```

Comment:
updating local variable just before return 0 ?

Reply:
This was actually an error. If I had sucessfully modified *len it would introduce too short data length. Removed this non necessary assingment.